advanced-game-design-team/Student Onboarding Guide.md

```diff
@@ -0,0 +1,406 @@
+# Splashkit Gamedev Cheat Sheet
+
+---
+
+Heeeeyyyyy. This guide will help you get started with game development using SplashKit's API. 
+
+It's important to follow the patterns here to the best of your ability throughout your completion of the capstone units:
+
+- It means your implementations will be consistent with other projects this team has worked / is working on, which will help you and your peers achieve more across more projects, and in less time.
+
+- It will help you reason about and debug your code far more easily, so you can spend more time having fun and less time confused and angry.
+
+- They're simple rules which are broadly good practice across the majority of software projects.
+
+---
+
+## Introduction
+
+**Let's define some concepts that this document emphasises:**
+
+### State
+
+**State** refers to all the variable values that represent your game's current condition. This includes positions of players and enemies, health levels, scores, game level, and any other data that changes as the game progresses.
+
+### Rendering
+
+**Rendering** is the process of drawing the current state on the screen. Rendering functions take the state and use it to produce visual representations. For example, if your state includes player coordinates in a normalized space (e.g., a range from 0 to 1), the rendering function can map these coordinates to actual screen positions. This makes handling resolution changes and different screen sizes easier.
+
+### Control Flow
+
+**Control Flow** dictates how your program executes, managing the sequence and timing of state updates and rendering. Control flow interacts with state by updating it based on player input or game events, and it interacts with rendering by calling rendering functions to update the visuals.
+
+---
+
+## 1. Separate State and Render Concerns
+
+**Principle**: Keep state logic and rendering logic separate.
+
+**Why?** This makes the code easier to manage and debug. When state and rendering logic are mixed, it becomes hard to track bugs and make changes. 
+
+Here are some examples:
+
+**Debugging Complexity:**
+
+If your rendering code modifies the game state, you might encounter visual bugs that are difficult to diagnose. For example, an animation function that inadvertently changes player coordinates can cause unpredictable visual outcomes. 
+
+*Example:*
+
+```cpp
+// Bad: Rendering logic mixed with game state updates in a larger context
```

Comment:
There's a bit of inconsistency with how you're presenting these examples. I personally think:

> *Example:*
> 
> ```cpp
> // Bad: Rendering logic mixed with game state updates in a larger context
> //code here
> ```

is more readable than:

> *Bad example:*
> 
> ```cpp
> //code here
> ```

But I think you should pick one of the two "good/bad example code" formats and commit to it instead of switching back and forth.